### ISSUE TITLE
GHCJS compilation

### ISSUE THREAD

**Issue by NCrashed** [haskell-servant/servant-multipart#48]
If you try to compile the package with GHCJS (for instance, for generation of frontend from servant API), you will get:
```
error: no C compiler provided for this platform
```

**Comment by NCrashed** [comment]
I have split package to `api` and backend part. https://github.com/hexresearch/servant-multipart/commit/92b6cb3458d51642e448862756d70d6df760e5b6

**Comment by tfc** [comment]
Hi, i could use this, too. Would you mind opening a pull request in this repo in order to upstream it?

**Comment by tfc** [comment]
@maksbotan @arianvp any way to get this in? i would also port the pull request to this repo if necessary.

**Comment by maksbotan** [comment]
Indeed, if for some weird reason you guys use `servant-multipart` with GHCJS (why tho?), it should be splitted into `servant-multipart-api`, `servant-multipart-client` and `servant-multipart` (containing server code).

I would appreciate a draft PR that I can improve on. Basically, please make the current package a subdir of the repo and add two more packages, like `servant` itself does it.

**Comment by tfc** [comment]
> Indeed, if for some weird reason you guys use `servant-multipart` with GHCJS (why tho?), it should be splitted into `servant-multipart-api`, `servant-multipart-client` and `servant-multipart` (containing server code).

Does anything speak against doing that? I look at this from a perspective where i have an API which simply contains file uploads and need to generate client code for it.

> I would appreciate a draft PR that I can improve on. Basically, please make the current package a subdir of the repo and add two more packages, like `servant` itself does it.

Makes sense.

**Comment by gdeest** [comment]
Just a quick heads-up here,

There was some discussion about merging most sub-projects, including `servant-multipart`, into the main servant repository (see: https://github.com/haskell-servant/servant/issues/1404 ). The main reason is that the cookbooks induce a dependency loop between the various projects, and updates introduce frequent breakage.

Maybe is this a good opportunity to do so for this project ? CI is already setup for GHCJS there.

**Comment by gdeest** [comment]
For the record, I am :+1: for the split.

**Comment by ysangkok** [comment]
I removed GHCjs support since it is too old. If we need CI with JavaScript, we should add that. But I think WASM might be more popular.
